notification/src/handler/notification/notification.handler.js

```diff
@@ -377,12 +377,23 @@ async function getTransactionTypeAndStateOrNull(notificationRequestItem) {
   )
   const adyenEventCode = notificationRequestItem.eventCode
   const adyenEventSuccess = notificationRequestItem.success
+  const adyenEventFailReason = notificationRequestItem.reason
 
   // eslint-disable-next-line max-len
-  const adyenEvent = _.find(
-    adyenEvents,
-    (e) => e.eventCode === adyenEventCode && e.success === adyenEventSuccess,
-  )
+  let adyenEvent = null;
+  _.find(adyenEvents, (e) => {
+    if (e.eventCode === adyenEventCode && e.success === adyenEventSuccess) {
+      if (adyenEventFailReason && e.reason === adyenEventFailReason) {
+        adyenEvent = e;
+        return true;
+      }
+      if (!adyenEvent) {
+        adyenEvent = e;
+      }
+    }
+    return false;
+  });
+  delete adyenEvent.reason
```

Comment:
why do we need to delete it? would it trigger some errors?

Reply:
no reason. we don't send `reason` for other notifications, that's all. i can send it if you think it would be helpful.

Reply:
it's unused in the code that references this method, so it's safe to keep